dotcom-rendering/src/web/experiments/lib/ab-localstorage.ts

```diff
@@ -0,0 +1,30 @@
+import type { Participations } from '@guardian/ab-core';
+import { storage } from '@guardian/libs';
+
+/**
+ * These utils have been lifted from the equivalent Frontend file
+ * https://github.com/guardian/frontend/blob/18f5d8aa4a4191f4efb58b37dc39685860369e04/static/src/javascripts/projects/common/modules/experiments/ab-local-storage.ts
+ */
+
+const participationsKey = 'gu.ab.participations';
+
+// -------
+// Reading
+// -------
+export const getParticipationsFromLocalStorage = (): Participations =>
+	(storage.local.get(participationsKey) as Participations | undefined) ?? {};
```

Comment:
Could we do a bit of parsing or validation there? What happens if the object is malformed instead of simply missing?

```ts
const isParticipations = (participations: unknown): participations is Participations => {
  if(!isObject(participations)) return false;
  if(Object.values(maybeParticipations).some(participation => !isString(participation.variant)) return false;
  return true;
}
```

Reply:
I've added in 👍